homeassistant/components/reolink/host.py

```diff
@@ -246,12 +246,18 @@ async def _renew(self) -> None:
 
     async def register_webhook(self) -> None:
         """Register the webhook for motion events."""
-        self.webhook_id = f"{DOMAIN}_{self.unique_id.replace(':', '')}"
+        self.webhook_id = f"{DOMAIN}_{self.unique_id.replace(':', '')}_ONVIF"
         event_id = self.webhook_id
 
-        webhook.async_register(
-            self._hass, DOMAIN, event_id, event_id, self.handle_webhook
-        )
+        try:
+            webhook.async_register(
+                self._hass, DOMAIN, event_id, event_id, self.handle_webhook
+            )
+        except ValueError:
```

Comment:
I think the problem is that we don't unregister the webhook before trying to set up the integration again via raising `ConfigEntryNotReady`.

Reply:
Before raising ConfigEntryNotReady we do call `host.stop` which unregisters the webhook:

https://github.com/home-assistant/core/blob/0c00d8b55c6592d329a5be1eee9212341887e872/homeassistant/components/reolink/__init__.py#L63-L64

https://github.com/home-assistant/core/blob/0c00d8b55c6592d329a5be1eee9212341887e872/homeassistant/components/reolink/host.py#L167-L169

https://github.com/home-assistant/core/blob/0c00d8b55c6592d329a5be1eee9212341887e872/homeassistant/components/reolink/host.py#L280-L284

Reply:
@MartinHjelmare do you have any other ideas?

Reply:
We don't do that when raising ConfigEntryNotReady from the first refresh of the coordinator.

Reply:
Not called here

```python
    except UserNotAdmin as err:
        raise ConfigEntryAuthFailed(err) from err
```

Reply:
> Not called here
> 
> ```python
>     except UserNotAdmin as err:
>         raise ConfigEntryAuthFailed(err) from err
> ```

@balloob UserNotAdmin is only raised here:
https://github.com/home-assistant/core/blob/0c00d8b55c6592d329a5be1eee9212341887e872/homeassistant/components/reolink/host.py#L79-L80
And we call stop() in the line before raising.
However this is a bit confusing so the already merged PR https://github.com/home-assistant/core/pull/87121 indeed removed the stop() there and put it in the line you quoted.

Reply:
@MartinHjelmare thank you so much, that is brilljant, never thought about that!
That will probably be it!